packages/backend/src/clients/EmailClient/EmailClient.ts

```diff
@@ -0,0 +1,89 @@
+import { SmptError } from '../../errors';
+import Logger from '../../logger';
+
+type Dependencies = {
+    lightdashConfig: Pick<LightdashConfig, 'smtp'>;
+};
+
+export default class EmailClient {
+    lightdashConfig: Pick<LightdashConfig, 'smtp'>;
+
+    transporter: nodemailer.Transporter | undefined;
+
+    constructor({ lightdashConfig }: Dependencies) {
+        this.lightdashConfig = lightdashConfig;
+
+        if (this.lightdashConfig.smtp) {
+            Logger.debug(`Create email transporter`);
+
+            const auth: AuthenticationType = this.lightdashConfig.smtp.auth
+                .accessToken
+                ? {
+                      type: 'OAuth2',
+                      user: this.lightdashConfig.smtp.auth.user,
+                      accessToken: this.lightdashConfig.smtp.auth.accessToken,
+                  }
+                : {
+                      user: this.lightdashConfig.smtp.auth.user,
+                      pass: this.lightdashConfig.smtp.auth.pass,
+                  };
+
+            this.transporter = nodemailer.createTransport(
+                {
+                    host: this.lightdashConfig.smtp.host,
+                    port: this.lightdashConfig.smtp.port,
+                    secure: this.lightdashConfig.smtp.port === 465, // false for any port beside 465, other ports use STARTTLS instead.
+                    auth,
+                    requireTLS: this.lightdashConfig.smtp.secure,
+                    tls: this.lightdashConfig.smtp.allowInvalidCertificate
+                        ? { rejectUnauthorized: false }
+                        : undefined,
+                },
+                {
+                    from: `"${this.lightdashConfig.smtp.sender.name}" <${this.lightdashConfig.smtp.sender.email}>`,
+                },
+            );
+
+            this.transporter.verify((error: any) => {
+                if (error) {
+                    throw new SmptError(
+                        `Failed to verify email transporter. ${error}`,
+                        {
+                            error,
+                        },
+                    );
+                } else {
+                    Logger.debug(`Email transporter verified with success`);
+                }
+            });
+        }
+    }
+
+    private async sendEmail(options: Mail.Options) {
+        if (this.transporter) {
+            try {
+                const info = await this.transporter.sendMail(options);
+                Logger.debug(`Email sent: ${info.messageId}`);
+            } catch (error) {
+                throw new SmptError(`Failed to send email. ${error}`, {
+                    error,
+                });
+            }
+        }
+    }
+
+    public async sendPasswordRecoveryEmail(link: PasswordResetLink) {
+        return this.sendEmail({
+            to: link.email,
+            subject: 'Reset Lightdash password',
+            text: `Reset your password here: ${
```

Comment:
In future we should have a directory of templates for rendering these emails rather than inline here